generator/generator.dart

```diff
@@ -118,22 +127,68 @@ String _hashToString(List<int> bytes) {
   return fileName;
 }
 
-void _generateFontsLists(Directory fontDirectory) {
-  List<String> currentFonts = File(_currentFontsPath).readAsLinesSync();
+// Utility class to track font family deltas.
+//
+// [fontDirectory] represents a possibly updated directory.
+class _FamiliesDelta {
+  _FamiliesDelta(Directory fontDirectory) {
+    _init(fontDirectory);
+  }
+
+  late final Set<String> added;
+  late final Set<String> removed;
+  late final Set<String> all;
 
-  List<String> addedFonts = [];
-  List<String> newCurrentFonts = [];
+  void _init(Directory fontDirectory) {
+    // Currently supported families.
+    Set<String> familiesSupported =
+        Set.from(File(_familiesSupportedPath).readAsLinesSync());
 
-  for (final item in fontDirectory.family) {
-    final family = item.name;
-    if (!currentFonts.contains(family)) {
-      addedFonts.add('* $family');
+    // Newly supported families.
+    all = Set<String>.from(
+      fontDirectory.family.map<String>((item) => item.name),
+    );
+
+    added = all.difference(familiesSupported);
+    removed = familiesSupported.difference(all);
+  }
+
+  // Printable list of supported font families.
+  String printableSupported() => all.join('\n');
+
+  // Printable diff of font families, suitable for markdown
+  // (e.g. CHANGELOG, PR description).
+  String printableDiff() {
```

Comment:
nit: s/printableDiff/markdownDiff or something to differentiate it from printableSupported and its intended use

Reply:
This is not really a nit, I think the public method names should be distinct for these 2 methods since they appear to be used for 2 different purposes